### PR TITLE
DP-1847: Add new client for managing webhook tokens with okdata-permission-api

### PR DESCRIPTION
**Pull request by oyvinyg**

### CODE REVIEW

File: okdata/sdk/webhook/client.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+          {
+            "token": "774d8f35-fb4b-4e06-9d7f-54d1a08589b4",
+            "created_by": "your-username",
+            "dataset_id": "some-dataset",
+            "service": "service-account-some-service",
+            "created_at": "2020-02-29T00:00:00+00:00",
+            "expires_at": "2022-02-28T00:00:00+00:00",
+            "is_active": True,
+          }
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:29]: Bra med eksempel. :+1:

File: okdata/sdk/webhook/client.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        Note!
+        The service_name parameter should NOT be prefixed with "service-account-"
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:31]: Slenge inn et punktum?

File: okdata/sdk/webhook/client.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    def authorize_webhook_token(self, dataset_id: str, token: str, retries: int = 0):
+        """Checks if a webhook token has access to perform operations on a dataset from a logged in service.
+
+        Returns a dictionary on the form:
+          {
+            "access": True
+            "reason": None
+          }
+        """
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:78]: Bør være på imperativsform: "Check if...", "Return a..." ([PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)).

File: okdata/sdk/webhook/client.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    def create_webhook_token(
+        self, dataset_id: str, service_name: str, retries: int = 0
+    ):
+        """Create a webhook token associated with a dataset and an okdata service.
+
+        Returns a dictionary on the form:
+
+          {
+            "token": "774d8f35-fb4b-4e06-9d7f-54d1a08589b4",
```
[the remark this answers is not in the document]
Comment (oyvinyg) [reply okdata/sdk/webhook/client.py:22]: Nei, random generert

File: okdata/sdk/webhook/client.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        self.__name__ = "permission"
+        super().__init__(config, auth, env)
+        self.api_url = f"{self.config.get('permissionApiUrl')}/webhooks"
+
+    def create_webhook_token(
+        self, dataset_id: str, service_name: str, retries: int = 0
+    ):
+        """Create a webhook token associated with a dataset and an okdata service.
+
+        Returns a dictionary on the form:
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:19]: "Return a ..." ([PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings))

File: okdata/sdk/webhook/client.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        """
+        url = f"{self.api_url}/{dataset_id}/tokens"
+        request_body = {"service": service_name}
+
+        return self.post(url, request_body, retries=retries).json()
+
+    def list_webhook_tokens(self, dataset_id: str, retries: int = 0):
+        """List all webhook tokens associated with a dataset.
+
+        Returns a list of dictionaries on the form:
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:41]: "Return a ..."

File: okdata/sdk/webhook/client.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+          .
+          ]
+        """
+        url = f"{self.api_url}/{dataset_id}/tokens"
+        return self.get(url, retries=retries).json()
+
+    def delete_webhook_token(self, dataset_id: str, token: str, retries: int = 0):
+        """Delete a webhook token.
+
+        Returns a dictionary on the form:
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:62]: "Return a ..."

File: okdata/sdk/webhook/client.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            "message": "Deleted token for dataset dataset_id"
+          }
+        """
+        url = f"{self.api_url}/{dataset_id}/tokens/{token}"
+        return self.delete(url, retries=retries).json()
+
+    def authorize_webhook_token(self, dataset_id: str, token: str, retries: int = 0):
+        """Check if a webhook token has access to perform operations on a dataset from a logged in service.
+
+        Returns a dictionary on the form:
```
Comment (simenheg) [inline okdata/sdk/webhook/client.py:73]: "Return a ..."

**Review by simenheg — APPROVED** [review]

**Pull request merged**